### PR TITLE
Provided GroupingPlugin with public methods to expand or collapse all…

### PR DESCRIPTION
**Pull request by FrankW76**
… groups

### CODE REVIEW

File: domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java [hunk 1]
```diff
@@ -96,6 +99,26 @@ public void appendRow(DataTable<T> dataTable, TableRow<T> tableRow) {
         }
     }
 
+    public void expandAllGroups() {
```
Comment (vegegoku) [inline domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java:102]: Change the name to `expandAll`

File: domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java [hunk 2]
```diff
@@ -96,6 +99,26 @@ public void appendRow(DataTable<T> dataTable, TableRow<T> tableRow) {
...
+        for (DataGroup<T> dataGroup : dataGroups.values()) {
+            if (!dataGroup.expanded)
+            {
```
Comment (vegegoku) [inline domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java:105]: Please stick to the project formatting here.
Maybe we should provide a format style in the project itself, for now please keep the open brace in the same if statement line.

File: domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java [hunk 3]
```diff
@@ -96,6 +99,26 @@ public void appendRow(DataTable<T> dataTable, TableRow<T> tableRow) {
...
+                dataGroup.toggleGroup();
+                dataGroup.getGroupIconSupplier().toggleIcon();
+            }
+        }
+    }
+
+    public void collapseAllGroups() {
+        for (DataGroup<T> dataGroup : dataGroups.values()) {
+            if (dataGroup.expanded)
+            {
```
Comment (vegegoku) [inline domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java:115]: same as above.

File: domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java [hunk 4]
```diff
@@ -118,13 +142,21 @@ public void toggleGroup() {
...
                     .toggleDisplay(expanded));
-
         }
 
         public void addRow(TableRow<T> tableRow) {
             groupRows.add(tableRow);
         }
 
+        private void setGroupIconSupplier(BaseIcon<?> groupIconSupplier)
+        {
```
Comment (vegegoku) [inline domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java:152]: same braces formatting here

File: domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java [hunk 5]
```diff
@@ -118,13 +142,21 @@ public void toggleGroup() {
...
+            this.groupIconSupplier = groupIconSupplier;
+        }
+
+        private BaseIcon<?> getGroupIconSupplier()
+        {
```
Comment (vegegoku) [inline domino-ui/src/main/java/org/dominokit/domino/ui/datatable/plugins/GroupingPlugin.java:157]: And here

**Comment by vegegoku** [comment]
sorry for the delay on this, please once the comments are resolved I will merge this PR.

**Comment by FrankW76** [comment]
Sorry for the braces. I am not used to put the braces behind the lines.
Maybe a checkstyle file would be handy ?

I will fix your comments tomorrow

**Pull request merged**